pkg/reconciler/service/resources/rolloutorchestrator_test.go

```diff
@@ -0,0 +1,1318 @@
+						TrafficTarget: servingv1.TrafficTarget{
+							RevisionName:   "service-001-00003",
+							LatestRevision: ptr.Bool(true),
+							Percent:        ptr.Int64(100),
+						},
+						MinScale: ptr.Int32(8),
+						MaxScale: ptr.Int32(10),
+					},
+				},
+				StageTarget: v1.StageTarget{
+					StageTargetRevisions: []v1.TargetRevision{
+						{
+							TrafficTarget: servingv1.TrafficTarget{
+								RevisionName:   "service-001-00002",
+								LatestRevision: ptr.Bool(false),
+								Percent:        ptr.Int64(20),
+							},
+							MinScale: ptr.Int32(8),
+							MaxScale: ptr.Int32(10),
+						},
+						{
+							TrafficTarget: servingv1.TrafficTarget{
+								RevisionName:   "service-001-00003",
+								LatestRevision: ptr.Bool(true),
+								Percent:        ptr.Int64(80),
+							},
+							MinScale: ptr.Int32(8),
+							MaxScale: ptr.Int32(10),
+						},
+					},
+					TargetFinishTime: testTime,
+				},
+			},
+		},
+	}, {
+		name: "Test the UpdateFinalTargetRev with a new final target",
+		ultimateTarget: []v1.TargetRevision{
+			{
+				TrafficTarget: servingv1.TrafficTarget{
+					RevisionName:   "service-001-00004",
+					LatestRevision: ptr.Bool(true),
+					Percent:        ptr.Int64(100),
+				},
+				MinScale: ptr.Int32(9),
+				MaxScale: ptr.Int32(19),
+			},
+		},
+		ro: &v1.RolloutOrchestrator{
+			Spec: v1.RolloutOrchestratorSpec{
+				TargetRevisions: []v1.TargetRevision{
+					{
+						TrafficTarget: servingv1.TrafficTarget{
+							RevisionName:   "service-001-00003",
+							LatestRevision: ptr.Bool(true),
+							Percent:        ptr.Int64(100),
+						},
+						MinScale: ptr.Int32(8),
+						MaxScale: ptr.Int32(10),
+					},
+				},
+				StageTarget: v1.StageTarget{
+					StageTargetRevisions: []v1.TargetRevision{
+						{
+							TrafficTarget: servingv1.TrafficTarget{
+								RevisionName:   "service-001-00002",
+								LatestRevision: ptr.Bool(false),
+								Percent:        ptr.Int64(20),
+							},
+							MinScale: ptr.Int32(8),
+							MaxScale: ptr.Int32(10),
+						},
+						{
+							TrafficTarget: servingv1.TrafficTarget{
+								RevisionName:   "service-001-00003",
+								LatestRevision: ptr.Bool(true),
+								Percent:        ptr.Int64(80),
+							},
+							MinScale: ptr.Int32(8),
+							MaxScale: ptr.Int32(10),
+						},
+					},
+					TargetFinishTime: testTime,
+				},
+			},
+		},
+		ExpectedResult: &v1.RolloutOrchestrator{
+			Spec: v1.RolloutOrchestratorSpec{
+				TargetRevisions: []v1.TargetRevision{
+					{
+						TrafficTarget: servingv1.TrafficTarget{
+							RevisionName:   "service-001-00004",
+							LatestRevision: ptr.Bool(true),
+							Percent:        ptr.Int64(100),
+						},
+						MinScale: ptr.Int32(9),
+						MaxScale: ptr.Int32(19),
+					},
+				},
+				StageTarget: v1.StageTarget{
+					StageTargetRevisions: nil,
+					TargetFinishTime:     apis.VolatileTime{},
+				},
+			},
+		},
+	}, {
+		name: "Test the UpdateFinalTargetRev with a new final target no status",
```

Comment:
Do we allow updating to a new final target if the previous stage is not finished yet?